src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberCompleteWork
+ * @flow
+ */
+
+'use strict';
+
+import type { ReactCoroutine, ReactYield } from 'ReactCoroutine';
+import type { Fiber } from 'ReactFiber';
+
+var ReactChildFiber = require('ReactChildFiber');
+var ReactTypesOfWork = require('ReactTypesOfWork');
+var {
+  IndeterminateComponent,
+  FunctionalComponent,
+  ClassComponent,
+  HostComponent,
+  CoroutineComponent,
+  YieldComponent,
+} = ReactTypesOfWork;
+
+function transferOutput(unitOfWork : Fiber) {
+  var child = unitOfWork.child;
+  if (child && !child.sibling) {
+    // Common case
+    unitOfWork.output = child.output;
+    return;
+  }
+  var output = [];
```

Comment:
I originally tried to make it a flattened linked list but figured it'd be simpler with a nested set. I suppose I could leave it as a nested linked list. That's very interesting. Good observation.